### PR TITLE
Migrate ACT_OPEN_GATE to the activity actor system

### PR DESCRIPTION
**Pull request by rsulli55**
#### Summary
SUMMARY: Infrastructure "Migrate ACT_OPEN_GATE to the activity actor system."

#### Purpose of change
Per issue #40013 the goal is to migrate all activities from activity_handlers.cpp to the new activity system added in #37450.

#### Testing
All tests from test suite passed locally.  Also tried opening gates/doors in game.

### CODE REVIEW

File: src/activity_actor.cpp [hunk 1]
```diff
@@ -374,6 +375,41 @@ std::unique_ptr<activity_actor> migration_cancel_activity_actor::deserialize( Js
     return migration_cancel_activity_actor().clone();
 }
 
+void open_gate_activity_actor::start( player_activity &act, Character & )
+{
+    act.moves_total = moves;
+    act.moves_left = moves;
+    act.placement = placement;
```
Comment (ifreund) [inline src/activity_actor.cpp:382]: We don't really need to set this as it never gets used.
```suggestion
```

Comment (rsulli55) [reply src/activity_actor.cpp:382]: Ok, sounds good.  I included that line as line 248 of gates.cpp had
`p.activity.placement = pos;` 
and I wasn't sure exactly where `player_activity::placement` was used. If we are removing that line, should we also just get rid of the `open_gate_activity_actor::placement` member variable?

Comment (ifreund) [reply src/activity_actor.cpp:382]: well, I think you'll need it to call `gates::open_gate()` in the finish function :D

Comment (rsulli55) [reply src/activity_actor.cpp:382]: Whoops, of course! And that explains where `player_activity::placement` was used. Thanks!

**Pull request merged**